test_framework/script.py

```diff
@@ -1096,6 +1106,7 @@ def from_desc(self, desc):
             self._decode_tree(desc[70:-1], parent=self.root)
         else:
             raise Exception
+        return self
```

Comment:
What do you think about just allowing `__init__()` to take an optional descriptor?

Reply:
I think its a good idea! Would it make sense to do this in another PR for all affected chapters? If you agree, I can track this in a dedicated issue.